src/Symfony/Component/Semaphore/composer.json

```diff
@@ -24,10 +24,12 @@
         "psr/log": "^1|^2|^3"
     },
     "require-dev": {
-        "predis/predis": "^1.1|^2.0"
+        "predis/predis": "^1.1|^2.0",
+        "symfony/lock": "^5.4 || ^6.0 || ^7.0"
     },
     "conflict": {
-        "symfony/cache": "<6.4"
+        "symfony/cache": "<6.4",
+        "symfony/lock": "<5.4 || >=8.0"
```

Comment:
why would it be incompatible with the nexts major versions of Lock?

Reply:
@jderusse we can not guaranty if lockfactory will look the same in 8.0 or not. We should always define optional dependeny constraints. This is same as if it is required with ^5.4 || ^6.0 || ^7.0

Reply:
We should even do the same for things like predis here

Reply:
Im sure fine to remove it but I would feel more save to keep it. Better save as sorry in the handling of dependencies.